build_tools/travis/install.sh

```diff
@@ -61,6 +61,8 @@ if [[ "$DISTRIB" == "conda" ]]; then
 
     conda create -n testenv --yes $TO_INSTALL
     source activate testenv
+    conda install pip --yes
```

Comment:
* pip is installed in any conda environment (at least by default) so you don't need this line
* can you only install pytest if $PYTHON_VERSION == "3.4" so that we can drop this special case once we stop supporting Python 3.4?
* can you use `pip install --upgrade --no-deps pytest`. Just a bit paranoid with `pip install -U` ...

Reply:
Yes, I was planning to those things, just wanted to make sure it was actually fixing it before diving putting time in it :-)

Will clean up now